sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobOutputStream.java

```diff
@@ -118,14 +115,8 @@ private BlobOutputStream(final BlobAsyncClient parentBlob) throws StorageExcepti
      *             An exception representing any error which occurred during the operation.
      */
     BlobOutputStream(final BlockBlobAsyncClient parentBlob, final BlobAccessConditions accessCondition) throws StorageException {
-        this((BlobAsyncClient) parentBlob);
-
-        this.accessCondition = accessCondition;
-        this.blockList = new TreeMap<>();
-        this.blockIdPrefix = UUID.randomUUID().toString() + "-";
-
-        this.streamType = BlobType.BLOCK_BLOB;
-        this.internalWriteThreshold = (int) BlockBlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE;
+        this(parentBlob, accessCondition, null, UUID.randomUUID().toString() + "-", new TreeMap<>(), 0,
+            BlobType.BLOCK_BLOB, BlockBlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE);
```

Comment:
The last argument is `BlockBlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE`, but the argument name is `internalWriteThreshold`. I can see this is what used to be there, but it is weird naming? // @alzimmermsft

Reply:
It's the threshold used until another page, block, etc is requested. Maybe this could be renamed to bufferSize? Could just remove the internal and have writeThreshold.